### PR TITLE
use File path getter over reading directly from property

### PR DESCRIPTION
**Pull request by SavePointSam**
In Nuclide, there is a RemoteFile. This object, while similar to Atom's File, doesn't have the exact same signature. However, they both share the same method `getPath`. Using this method in favor of reading the property directly, repairs prettier-atom's ability to 'format on save' when a user is working with remote files.

Fixes [facebook/nuclide#1443](https://github.com/facebook/nuclide/issues/1443) and [prettier/prettier-atom#322](https://github.com/prettier/prettier-atom/issues/322).

___

Due to an issue with the yarn/npm registry at the moment. I am unable to successfully install dependencies. Various babel packages seem to randomly 404. Therefore, I'm unable to run all the required scripts outlined in CONTRIBUTING.md. What would you like me to do in terms of running tests and adding myself to the contributors list?

I'm also unfamiliar with Flow. I attempted to update the `TextEditor` type definition, but was unsure what needed to happen and opted to avoid modifying it as it exists now. Can someone help me?

### CODE REVIEW

File: src/editorInterface/index.js [hunk 1]
```diff
@@ -42,7 +42,8 @@ const isCurrentScopeMarkdownScope = (editor: TextEditor) =>
 
 const isCurrentScopeVueScope = (editor: TextEditor) => getVueScopes().includes(getCurrentScope(editor));
 
-const getCurrentFilePath = (editor: TextEditor) => (editor.buffer.file ? editor.buffer.file.path : undefined);
+const getCurrentFilePath = (editor: TextEditor) =>
+  editor.buffer.file ? editor.buffer.file.getPath() : undefined;
```
Comment (darahak) [inline src/editorInterface/index.js]: To fix the Flow error on this line (also reported in CI), you need to update the type defs in [decls/index.js](https://github.com/prettier/prettier-atom/blob/master/decls/index.js).

The [documentation](https://atom.io/docs/api/v1.23.3/File#instance-getPath) indicates that `getPath()` returns a `String`, so the `TextEditor` type should be updated like this:

```js
declare type TextEditor = {
  getGrammar: () => { scopeName: string },
  getBuffer: () => { getRange: () => Range, setTextViaDiff: (text: string) => Range },
  getCursorScreenPosition: () => Point,
  getLastCursor: () => { getScopeDescriptor: () => Atom$ScopeDescriptor },
  getSelectedText: () => string,
  getSelectedBufferRanges: () => Ranges,
  getTextInBufferRange: (bufferRange: Range) => string,
  setCursorScreenPosition: (point: Point) => Point,
  setTextInBufferRange: (bufferRange: Range, text: string) => Range,
  buffer: {
    file: ?{
      path: ?FilePath,
      getPath: () => string // HERE
    }
  },
  backwardsScanInBufferRange: (regex: RegExp, Range: Range, iterator: Atom$Iterator) => void,
};
```

**Comment by darahak** [comment]
Thank you for your contribution!

To complete the PR, you will also need to do this:

- [ ] Amend your commit message and follow this template (for the automatic **changelog** updates): `type(scope): message` (see previous [commits](https://github.com/prettier/prettier-atom/commits/master))
- [ ] Run the `build` script and commit the modifications in `/dist`.
- [ ] Run the `addContributor` script if you want to add yourself to the contributor list and commit the changes.

File: src/editorInterface/index.js (hunk 1 above)
Comment (SavePointSam) [reply src/editorInterface/index.js]: Thank you!

**Comment by SavePointSam** [comment]
I've gone back and updated my commit message, add the type declaration and was able to add my contributing information. However, I'm unable to run a build as I can't download the dependencies still.

```
~/source/prettier-atom issue-322_nuclide-remote*
❯ yarn
yarn install v1.3.2
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
error An unexpected error occurred: "https://registry.yarnpkg.com/@babel/types/-/types-7.0.0-beta.31.tgz: Request failed \"404 Not Found\"".
info If you think this is a bug, please open a bug report with the information provided in "/Users/savepointsam/source/prettier-atom/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

What would you like to do about this?

**Comment by SavePointSam** [comment]
Alright, I was finally able to install the dependencies. I'll finish up shortly. Apologies for all the commits.

**Comment by SavePointSam** [comment]
Cool, all done!

**Comment by robwise** [comment]
Thanks for the PR!

**Pull request merged**